### PR TITLE
cow body weight update

### PR DESCRIPTION
**Pull request by manfei-L**
This commit is about updating on the cow BW change. In this change, the lost BW during the lactation period due to lactation is restored during the dry period, and the pregnancy status adjusts the ADG for 1st and 2nd lactation. 
The complement of culling with the death part is also included in this commit.

### CODE REVIEW

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 1]
```diff
@@ -357,29 +359,46 @@ def get_bw_change(self, CI):
...
+            else:
+                target_adg_cow = \
+                    (self.mature_body_weight - self.body_weight)/(self.gestation_length - self.days_in_preg+1)
+
         else:  # parity > 2
             target_adg_cow = 0
 
         if self.days_in_milk == 0:
-            bodyweight_tissue = 0
+            if self.days_in_preg == 0:
```
Comment (KFosterReed) [inline RUFAS/routines/animal/life_cycle/cow.py:383]: What cows have 0 days in milk and 0 days in pregnancy?

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 2]
```diff
@@ -357,29 +359,46 @@ def get_bw_change(self, CI):
...
+                bodyweight_tissue = 0
+            else:
+                bodyweight_tissue = self.tissue_changed / (self.gestation_length - AnimalBase.config['days_in_preg_when_dry'])
```
Comment (KFosterReed) [inline RUFAS/routines/animal/life_cycle/cow.py:386]: Also, I don't see where self.tissue_changed is updated to be non-zero before this. Is this from the day before?

**Comment by KFosterReed** [comment]
Hi Manfei, I really like the flow chart that you have as part of the documentation for this section. I have a couple of questions: 
1) Can you please also link to the full pseudocode as part of the documentation for commit and include a description of the culling by death update
2) Can you please review the equations in the flow chart? Some of the tissue changes are not the same as the ones in the code. Thanks!

**Comment by militsasotirova** [comment]
I'm not sure if this was happening before the merge of master into this branch or not, but I'm getting the following error when I run (I think it happens with every seed but I know it happens on seed 0). I think this is an issue where the JSON files weren't properly updated in the merge?
Screen Shot 2021-04-26 at 8 16 02 PM

File: RUFAS/routines/animal/life_cycle/cow.py (hunk 1 above)
Comment (manfei-L) [reply RUFAS/routines/animal/life_cycle/cow.py:383]: The cow on the calving day would have DIM and DIP both equals to 0.

File: RUFAS/routines/animal/life_cycle/cow.py (hunk 2 above)
Comment (manfei-L) [reply RUFAS/routines/animal/life_cycle/cow.py:386]: Tissue changed is calculated on the last day of milking, and it is used to calculate the rate of restore during the dry period, which is a constant (linear gaining).

**Comment by KFosterReed** [comment]
> I'm not sure if this was happening before the merge of master into this branch or not, but I'm getting the following error when I run (I think it happens with every seed but I know it happens on seed 0). I think this is an issue where the JSON files weren't properly updated in the merge?
> Screen Shot 2021-04-26 at 8 16 02 PM

I think this is my fault. I was trying to resolve the conflicts yesterday but since I couldn't figure out the commit from my Pycharm, I tried to resolve it in Github and accepted the new animal_management.json rather than the master branch one. @manfei-L can you please resolve this? For future pull requests, please remember to update all .json files as well before submitting the pull request.

Thanks!

File: RUFAS/routines/animal/life_cycle/cow.py (hunk 2 above)
Comment (KFosterReed) [reply RUFAS/routines/animal/life_cycle/cow.py:386]: Ok, for user readability, I would like to change the order of these if else statements to have the lactation tissue changes come first in the order of the code followed by the dry cow calculation. Also, please add comments to each section with a short statement identifying what each section is doing

File: RUFAS/routines/animal/life_cycle/cow.py (hunk 1 above)
Comment (KFosterReed) [reply RUFAS/routines/animal/life_cycle/cow.py:383]: I think my question then is do we need both of these if statements? I think there is a way to streamline the logic flow in this function.

**Pull request closed without merging**